supabase/functions/create-checkout/index.ts

```diff
@@ -0,0 +1,108 @@
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.39.0'
+import Stripe from 'https://esm.sh/stripe@14.5.0'
+import { corsHeaders } from '../_shared/cors.ts'
+
+const stripe = new Stripe(Deno.env.get('STRIPE_SECRET_KEY') || '', {
+  apiVersion: '2025-02-24.acacia',
+  httpClient: Stripe.createFetchHttpClient()
+})
+
+Deno.serve(async (req) => {
+  if (req.method === 'OPTIONS') {
+    return new Response('ok', { headers: corsHeaders })
+  }
+
+  try {
+    const authHeader = req.headers.get('Authorization')
+    if (!authHeader) {
+      throw new Error('No authorization header')
+    }
+
+    const supabase = createClient(
+      Deno.env.get('SUPABASE_URL') ?? '',
+      Deno.env.get('SUPABASE_ANON_KEY') ?? '',
+      {
+        global: {
+          headers: { Authorization: authHeader }
+        }
+      }
+    )
+
+    const token = authHeader.replace('Bearer ', '')
+    const {
+      data: { user },
+      error: userError,
+    } = await supabase.auth.getUser(token)
+
+    if (userError || !user) {
+      console.error('Supabase user error:', userError)
+      throw new Error('Error getting user')
+    }
+
+    const { data: existingLicense } = await supabase
```

Comment:
We might need to figure out what happens with the code if they have multiple licenses

Reply:
nvm, this is just being used to check if they have any existing license